cmd/restic/cmd_check.go

```diff
@@ -252,12 +253,19 @@ func runCheck(opts CheckOptions, gopts GlobalOptions, args []string) error {
 
 	Verbosef("check snapshots, trees and blobs\n")
 	errChan = make(chan error)
+	var wg sync.WaitGroup
+
+	wg.Add(1)
 	go func() {
+		defer wg.Done()
 		bar := newProgressMax(!gopts.Quiet, 0, "snapshots")
 		defer bar.Done()
 		chkr.Structure(gopts.ctx, bar, errChan)
 	}()
 
+	// Wait for the progress bar to be complete before printing more below.
+	wg.Wait()
```

Comment:
This will deadlock if the check outputs errors. The correct location is after the `for err := range errChan {`-loop.